packages/auth-helpers/src/lib/siwe/create-siwe-message.ts

```diff
@@ -0,0 +1,90 @@
+import { SiweMessage } from 'siwe';
+import { BaseSiweMessage, CapacityDelegationFields } from '@lit-protocol/types';
+import {
+  LitAbility,
+  WithCapacityDelegation,
+  WithRecap,
+} from '@lit-protocol/types';
+import { LitRLIResource } from '../resources';
+
+import {
+  createCapacityCreditsResourceData,
+  addRecapToSiweMessage,
+} from './siwe-helper';
+
+export const createSiweMessage = async <T extends BaseSiweMessage>(
+  params: T
+): Promise<string> => {
+  // -- validations
+  if (!params.walletAddress) {
+    throw new Error('walletAddress is required');
+  }
+
+  const ONE_WEEK_FROM_NOW = new Date(
+    Date.now() + 1000 * 60 * 60 * 24 * 7
+  ).toISOString();
+
+  let siweParams = {
+    domain: params?.domain ?? 'localhost',
+    address: params.walletAddress,
+    statement:
+      params?.statement ??
+      'This is a test statement.  You can put anything you want here.',
+    uri: params?.uri ?? 'https://localhost/login',
+    version: params?.version ?? '1',
+    chainId: params?.chainId ?? 1,
+    nonce: params.nonce,
+    expirationTime: params?.expiration ?? ONE_WEEK_FROM_NOW,
+  };
+
+  let siweMessage = new SiweMessage(siweParams);
+
+  // -- create a message with capacity credits
+  if (
+    'dAppOwnerWallet' in params || // required param
+    'uses' in params || // optional
+    'delegateeAddresses' in params || // optional
+    'capacityTokenId' in params // optional
```

Comment:
Can't we use types here from `BaseSiweMessage`?

Reply:
No, typescript types cannot be used as strings in the code